File: artifacts/api-server/src/routes/health.ts
import { Router, type IRouter } from "express";
import {
  HealthCheckResponse,
  RailzenHealthResponse,
} from "@workspace/api-zod";

const router: IRouter = Router();

router.get("/healthz", (_req, res) => {
  const data = HealthCheckResponse.parse({ status: "ok" });
  res.json(data);
});

router.get("/health", (_req, res) => {
  const data = RailzenHealthResponse.parse({
    status: "ok",
    service: "RailZen AI",
    phase: "0.1",
  });
  res.json(data);
});

export default router;

File: backend/app/api/routes/health.py
"""Health endpoints for local and proxied service checks."""

from fastapi import APIRouter

from ...schemas.health import HealthStatus

router = APIRouter(tags=["health"])


@router.get("/health")
def health() -> HealthStatus:
    return HealthStatus(status="ok", service="RailZen AI", phase="0.1")


@router.get("/healthz")
def healthz() -> dict[str, str]:
    return {"status": "ok"}